libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/HttpClient.kt

```diff
@@ -0,0 +1,70 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation
+
+import androidx.annotation.Keep
+import com.mapbox.navigator.HttpCode
+import com.mapbox.navigator.HttpInterface
+import com.mapbox.navigator.HttpResponse
+import com.mapbox.services.android.navigation.BuildConfig
+import java.io.ByteArrayOutputStream
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.logging.HttpLoggingInterceptor
+import okio.Okio
+import timber.log.Timber
+
+@Keep
+internal class HttpClient(
+    private val userAgent: String,
+    private val acceptGzipEncoding: Boolean = false,
+    private val clientBuilder: OkHttpClient.Builder = OkHttpClient.Builder()
+) : HttpInterface() {
+
+    private val client: OkHttpClient by lazy {
+        if (BuildConfig.DEBUG) {
+            val interceptor = HttpLoggingInterceptor(HttpLoggingInterceptor.Logger { message ->
+                Timber.d(message)
+            }).setLevel(HttpLoggingInterceptor.Level.BASIC)
+
+            clientBuilder.addInterceptor(interceptor)
+        }
+
+        clientBuilder.build()
+    }
+
+    init {
+        check(userAgent.isNotEmpty()) {
+            "Empty UserAgent is not allowed"
```

Comment:
Could we extract all magic numbers in `HttpClient` into constants?

Reply:
fixed